bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/SWTEventListener.java

```diff
@@ -29,5 +29,6 @@
  * </p>
  * @noreference This interface is not intended to be referenced by clients.
  */
+@Deprecated(forRemoval=true, since="2025-03")
```

Comment:
I think the since tag should represent the bundle version since which this is deprecated, which will probably be `3.129`.
Furthermore, since it's the hard to know for ordinary users when the API will actually be removed, I experimeted with a corresponding extension in PDE and I think that's helpful.
https://github.com/eclipse-pde/eclipse.pde/blob/026b764ff7b262ffb0ed5a3732ddd66ff4a89673/ui/org.eclipse.pde.core/src/org/eclipse/pde/core/plugin/PluginRegistry.java#L67

```suggestion
@Deprecated(forRemoval = true, since = "3.129 (removal in 2027-03 or later)")
```

Reply:
This has been discussed some time ago and decided to go with "human readable calendar form" (see https://github.com/eclipse-platform/eclipse.platform/blob/master/docs/Eclipse_API_Central_Deprecation_Policy.md#java-api ) as the bundle version is not easily translatable to release version and for some bundles it would even happen that same version will be shipped in multiple versions of the SDK thus making the 2 years timeframe calculation vague.

Reply:
If you have good idea how to improve it please bring your proposal to the next PMC call for a change. Till we get a change in guidelines I would prefer that we stick to them.

Reply:
> as the bundle version is not easily translatable to release version and for some bundles it would even happen that same version will be shipped in multiple versions of the SDK thus making the 2 years timeframe calculation vague.

Understand. The goal is the same and additionally it should be make simple for those that don't know our 2 year policy.

> If you have good idea how to improve it please bring your proposal to the next PMC call for a change.

Sure I'll bring this up in December. 👍🏽